src/main/java/se/vandmo/dependencylock/maven/Artifact.java

```diff
@@ -1,18 +1,45 @@
 package se.vandmo.dependencylock.maven;
 
-import static java.util.Objects.requireNonNull;
-import static java.util.Optional.ofNullable;
+import com.google.common.io.BaseEncoding;
+import com.google.common.io.Files;
+import org.apache.maven.model.Dependency;
+import org.apache.maven.plugin.logging.Log;
+import org.apache.maven.plugin.logging.SystemStreamLog;
 
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
 import java.util.Objects;
-import org.apache.maven.model.Dependency;
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.Optional.ofNullable;
 
 public final class Artifact implements Comparable<Artifact> {
+  private static final Log LOGGER = new SystemStreamLog();
+  private static final MessageDigest SHA256_DIGEST;
+
+  /**
+   * The current algorithm that is being used. If this doesn't match an error will be thrown and dependency files need
+   * to be regenerated.
+   */
+  public static final String ALGORITHM_HEADER = "sha256:";
+
+  static {
+    try {
+      SHA256_DIGEST = MessageDigest.getInstance("SHA-256");
```

Comment:
SHA384 ? SHA512?
Is SHA256 secure enough?

Reply:
Will change to 512